packages/atlas/src/joystream-lib/extrinsics.ts

```diff
@@ -465,7 +469,105 @@ export class JoystreamLibExtrinsics {
       block,
     }
   }
+  /*
+    === Channel payouts ===
+  */
+
+  getClaimableReward = async (
+    channelId: string,
+    cumulativeRewardEarned: StringifiedNumber | null,
+    nodeEndpoint: string,
+    payloadDataObjectId: string
+  ) => {
+    const payoutProof = await channelPayoutProof(
+      'URL',
+      `${nodeEndpoint}/api/v1/files/${payloadDataObjectId}`,
+      Number(channelId)
+    )
+
+    const cashout = new BN(payoutProof.cumulativeRewardEarned).sub(new BN(cumulativeRewardEarned || 0))
+    return cashout.toString()
+  }
+
+  claimRewardTx = async (
+    channelId: string,
+    memberId: MemberId,
+    cumulativeRewardEarned: StringifiedNumber | null,
+    nodeEndpoint: string,
+    payloadDataObjectId: string,
+    commitment: string
+  ) => {
+    await this.ensureApi()
 
+    const payoutProof = await channelPayoutProof(
+      'URL',
+      `${nodeEndpoint}/api/v1/files/${payloadDataObjectId}`,
+      Number(channelId)
+    )
+
+    const maxCashoutAllowed = await this.api.query.content.maxCashoutAllowed()
+    const minCashoutAllowed = await this.api.query.content.minCashoutAllowed()
+    const cashout = new BN(payoutProof.cumulativeRewardEarned).sub(new BN(cumulativeRewardEarned || 0))
+    const isPayoutProofVerified = verifyChannelPayoutProof(payoutProof) === commitment
+
+    if (maxCashoutAllowed.lt(cashout)) {
+      throw new JoystreamLibError({
+        name: 'FailedError',
+        message: `Channel cashout amount is too low to be claimed`,
+      })
+    }
+    if (minCashoutAllowed.gt(cashout)) {
+      throw new JoystreamLibError({
+        name: 'FailedError',
+        message: `Channel cashout amount is too low to be claimed`,
+      })
+    }
+    if (!isPayoutProofVerified) {
+      throw new JoystreamLibError({
+        name: 'FailedError',
+        message: `Incorrect payout proof`,
+      })
+    }
+    const pullPayment = createType('PalletContentPullPaymentElement', {
+      channelId: new BN(channelId),
+      cumulativeRewardEarned: new BN(payoutProof.cumulativeRewardEarned),
+      reason: u8aToHex(Buffer.from(payoutProof.reason, 'hex')),
+    })
+    const merkleBranch: PalletCommonMerkleTreeProofElementRecord[] = []
+    payoutProof.merkleBranch.forEach((m) => {
+      const proofElement = createType('PalletCommonMerkleTreeProofElementRecord', {
+        hash_: u8aToHex(Buffer.from(m.hash, 'hex')),
+        side: m.side ? { Right: null } : { Left: null },
+      })
+      merkleBranch.push(proofElement)
+    })
```

Comment:
I think this code was included in `@joystream/js` after all so we may not need it in our codebase. Can you check?